earth_enterprise/src/common/performancelogger.cpp

```diff
@@ -18,19 +18,139 @@
 #include <unistd.h>
 #include <sys/syscall.h>
 #include <time.h>
-
+#include <vector>
 #include "common/performancelogger.h"
 #include "common/notify.h"
 #include "common/timeutils.h"
 
 using namespace std;
 using namespace getime;
 
-// Initialize static members of Profiler class
-PerformanceLogger * const PerformanceLogger::_instance = new PerformanceLogger();
+std::ofstream ioPrefFile;
+std::string ioFileName;
+std::ofstream timePrefFile;
+std::ofstream threadPrefFile;
+
+void checkForLeadingZero(
+        std::stringstream& io,
+        std::stringstream& thread,
+        std::stringstream& time,
+        const int& val) {
+    if (val > 9) return;
+    std::string temp("0");
+    io << temp;
+    thread << temp;
+    time << temp;
+}
+
+void printToSStream(
+        std::stringstream& io,
+        std::stringstream& thread,
+        std::stringstream& time,
+        const char& delimiter,
+        const int& val) {
+    io << val << delimiter;
+    thread << val << delimiter;
+    time << val << delimiter;
+}
+
+void openFiles() {
+    time_t t = time(0);
+    tm date = *localtime(&t);
+    std::stringstream io,time,thread;
+    io << "io.";
+    time << "time.";
+    thread << "thread.";
+    checkForLeadingZero(io,thread,time,date.tm_mon);
+    printToSStream(io,thread,time,'-',date.tm_mon);
+    checkForLeadingZero(io,thread,time,date.tm_mday);
+    printToSStream(io,thread,time,'-',date.tm_mday);
+    printToSStream(io,thread,time,'-',date.tm_year+1900);
+    checkForLeadingZero(io,thread,time,date.tm_hour);
+    printToSStream(io,thread,time,':',date.tm_hour);
+    checkForLeadingZero(io,thread,time,date.tm_min);
+    printToSStream(io,thread,time,':',date.tm_min);
+    checkForLeadingZero(io,thread,time,date.tm_sec);
+    printToSStream(io,thread,time,'.',date.tm_sec);
+    io << "csv";
+    thread << "csv";
+    time << "csv";
+    // if not open, open and add header
+    if (!ioPrefFile.is_open()) { 
+      ioPrefFile.open(io.str().c_str());
```

Comment:
Minor: You don't need a C string to open a file.  A `std::string` works fine.

Reply:
I believe that is only the case in C++11